hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-applicationhistoryservice/src/main/java/org/apache/hadoop/yarn/server/applicationhistoryservice/ApplicationHistoryManagerOnTimelineStore.java

```diff
@@ -498,17 +498,19 @@ private ApplicationReportExt generateApplicationReport(TimelineEntity entity,
       if (app.appReport.getCurrentApplicationAttemptId() != null) {
         ApplicationAttemptReport appAttempt =
             getApplicationAttempt(app.appReport.getCurrentApplicationAttemptId());
-        if (appAttempt != null) {
```

Comment:
no, the idea here is to send an ApplicationAttemptNotFoundException exception adn to catch it settings appropriate value to the appReport
Look at getApplicationAttempt